syncd/syncd_applyview.cpp

```diff
@@ -672,10 +965,314 @@ std::shared_ptr<SaiObj> findCurrentBestMatchForRouterInterface(
             continue;
         }
 
-        // TODO we need special case for LOOPBACK and for matching OID's
-
         if (hasEqualAttribute(c, t, SAI_ROUTER_INTERFACE_ATTR_TYPE))
         {
+            if (hasEqualAttribute(curv, tmpv, c, t, SAI_ROUTER_INTERFACE_ATTR_VIRTUAL_ROUTER_ID))
+            {
+                int32_t type = t->getAttr(SAI_ROUTER_INTERFACE_ATTR_TYPE)->getSaiAttr()->value.s32;
+
+                if (type == SAI_ROUTER_INTERFACE_TYPE_PORT)
+                {
+                    /*
+                     * When type is PORT, then we need to check port ID as well.
+                     */
+
+                    if (hasEqualAttribute(curv, tmpv, c, t, SAI_ROUTER_INTERFACE_ATTR_PORT_ID))
+                    {
+                        return c;
+                    }
+                }
+                else if (type == SAI_ROUTER_INTERFACE_TYPE_LOOPBACK)
+                {
+                    /*
+                     * When type is defined as LOOPBACK, then PORT attribute is
+                     * not present If multiple loopbacks will be defined then
+                     * this is first that we hit, so to make it better we need
+                     * then extend this logic to look deeper into dependency
+                     * graph and find similarities.
+                     */
+
+                    return c;
+                }
+            }
+        }
+    }
+
+    return nullptr;
+}
+
+std::shared_ptr<SaiObj> findCurrentBestMatchForNextHop(
```

Comment:
because function searches for current existing object best match so it's more explaining itself
anyway this code will be removed in final version when i implement generic logic